### PR TITLE
feat(published_time_publisher): add unit test

### PR DESCRIPTION
**Pull request by brkay54**
## Description
**Part of:**

- https://github.com/autowarefoundation/autoware.universe/issues/6255

**Required from:**

- https://github.com/autowarefoundation/autoware.universe/pull/6440#pullrequestreview-1931367141

**Depends on:**

- https://github.com/autowarefoundation/autoware.universe/pull/6490
_because of renaming of `publish()` function_

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/tier4_autoware_utils/test/src/ros/test_published_time_publisher.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+TEST_F(PublishedTimePublisherTest, PublishMsgWithHeader)
+{
+  std_msgs::msg::Header header;
+  header.stamp = rclcpp::Time(1234);
+
+  // Use Published Time Publisher with a timestamp
+  published_time_publisher_->publish(test_publisher_, header);
+  rclcpp::spin_some(node_);
+
```
Comment (xmfcx) [inline common/tier4_autoware_utils/test/src/ros/test_published_time_publisher.cpp:65]:
```suggestion

ASSERT_TRUE(published_time_ != nullptr);

```

File: common/tier4_autoware_utils/test/src/ros/test_published_time_publisher.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+TEST_F(PublishedTimePublisherTest, PublishMsgWithTimestamp)
+{
+  std_msgs::msg::Header header;
+  header.stamp = rclcpp::Time(4321);
+
+  // Use Published Time Publisher with a timestamp
+  published_time_publisher_->publish(test_publisher_, header.stamp);
+  rclcpp::spin_some(node_);
+
```
Comment (xmfcx) [inline common/tier4_autoware_utils/test/src/ros/test_published_time_publisher.cpp:78]:
```suggestion

ASSERT_TRUE(published_time_ != nullptr);

```

File: common/tier4_autoware_utils/test/src/ros/test_published_time_publisher.cpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  void SetUp() override
+  {
+    ASSERT_TRUE(rclcpp::ok());
+
+    // Create a node
+    node_ = std::make_shared<rclcpp::Node>("PublishedTimePublisher_test_node");
+    // Create a publisher
+    test_publisher_ =
+      node_->create_publisher<std_msgs::msg::Header>("PublishedTimePublisher_test_topic", 1);
```
Comment (xmfcx) [inline common/tier4_autoware_utils/test/src/ros/test_published_time_publisher.cpp:41]:
```cpp
    // Simplify node and topic names for brevity and uniqueness
    test_name_ = ::testing::UnitTest::GetInstance()->current_test_info()->name();
    std::string base_name = "pub_time_test_" + test_name_; // Base name for node and topics

    // Create a node
    node_ = std::make_shared<rclcpp::Node>(base_name + "_node");
    // Create a publisher
    test_publisher_ = node_->create_publisher<std_msgs::msg::Header>(base_name + "_topic", 1);
```

Could you try something like this to get a unique node/topic name?

**Comment by xmfcx** [comment]
Thanks for the work❇️
Once the build-and-test-differential jobs pass, we can merge this.
It checks most of its functionality.

**Comment by brkay54** [comment]
@xmfcx I implemented some of the the automated review suggestions. I think it is ready to merge.

**Review by takayuki5168 — APPROVED** [review]

**Comment by brkay54** [comment]
@takayuki5168 Thank you 🙏🏼

**Pull request merged**